doc/examples/segmentation/plot_rolling_ball.py

```diff
@@ -0,0 +1,151 @@
+volume of the gel image umbra but is constrained by the upper surface of the
+umbra."
+
+.. [1] Sternberg, Stanley R. "Biomedical image processing." Computer 1 (1983):
+    22-34. :DOI:`10.1109/MC.1983.1654163`
+
+To implement this algorithm in skimage we can use the following:
+"""
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+from skimage import morphology
+from skimage import data
+from skimage.util import invert
+
+
+def rolling_ball(image, radius=30, white_background=False):
+    working_img = image.copy()
+    if white_background:
+        working_img = invert(working_img)
+
+    # tensor representation of the image
+    # (stacked blocks as described in the paper)
+    intensity = np.arange(np.iinfo(working_img.dtype).max)
+    background = (intensity[np.newaxis,
+                            np.newaxis, :] < working_img[..., np.newaxis])
+
+    # roll the ball around
+    # here: implemented as a sequence of small balls (B1, ..., Bk)
+    # because the current implementation of a closure with a large ball
+    # is slow and consumes a lot of memory
+    selem = morphology.ball(1)
+    for idx in range(radius):
+        background = morphology.erosion(background, selem)
+
+    for idx in range(radius):
+        background = morphology.dilation(background, selem)
+
+    # flatten the tensor
+    background = np.argmax(background == False, axis=-1)
+
+    filtered_image = working_img - background
+
+    if white_background:
+        filtered_image = invert(filtered_image)
+        background = invert(background)
+
+    return filtered_image, background
+
+
+######################################################################
+# And then test it using an image with white background
+
+
+image = data.page()
+
+filtered_image, background = rolling_ball(image, radius=30,
+                                          white_background=True)
+fig, ax = plt.subplots(nrows=1, ncols=3)
+
+ax[0].imshow(image, cmap='gray')
+ax[0].set_title('Original image')
+
+ax[1].imshow(background, cmap='gray')
+ax[1].set_title('Background')
+
+ax[2].imshow(filtered_image, cmap='gray')
+ax[2].set_title('Result')
+
+fig.tight_layout()
+plt.show()
+
+######################################################################
+# And with an image with black background
+
+image = data.coins()
+
+filtered_image, background = rolling_ball(image, radius=50)
+fig, ax = plt.subplots(nrows=1, ncols=3)
+
+ax[0].imshow(image, cmap='gray')
+ax[0].set_title('Original image')
+
+ax[1].imshow(background, cmap='gray')
+ax[1].set_title('Background')
+
+ax[2].imshow(filtered_image, cmap='gray')
+ax[2].set_title('Result')
+
+fig.tight_layout()
+plt.show()
+
+######################################################################
+# As you may have noticed, above implementation is rather slow.
+# If runtime is a concern, we can implement an efficient approximation
+# using a tophat filter and a disk element
+
+
+def rolling_ball(image, radius=50, white_background=False):
```

Comment:
maybe call this `rolling_disk`?

Reply:
nice suggestion; thanks! I changed the name.